website/content/en/blog/log-namespacing.md

```diff
@@ -0,0 +1,156 @@
+represents (eg: was the `timestamp` field generated by vector when it was ingested, or is it when
+the source originally created the event) but it can easily cause data collisions.
+
+Log namespacing also unblocks powerful features being worked on, such as end-to-end type checking
+of events in Vector.
+
+
+## How to enable
+
+The [global config] `schema.log_namespace` can be set to `true` to enable the new
+Log Namespacing feature for all components. The default is `false`.
+
+Every source also has a `log_namespace` config option. This will override the global setting,
+so you can try out Log Namespacing on individual sources.
+
+The following example enables the `log_namespace` feature globally, then disables it for a single
+source.
+
+```toml
+schema.log_namespace = true
+
+[sources.input_with_log_namespace]
+type = "demo_logs"
+format = "shuffle"
+lines = ["input_with_log_namespace"]
+interval = 1
+
+[sources.input_without_log_namespace]
+type = "demo_logs"
+format = "shuffle"
+lines = ["input_without_log_namespace"]
+interval = 1
+log_namespace = false
+
+[sinks.console]
+type = "console"
+inputs = ["input_with_log_namespace", "input_without_log_namespace"]
+encoding.codec = "json"
+
+```
+
+## How It Works
+
+
+### Data Layout
+
+There are three distinct types of data that Vector handles.
+
+(Examples are from the `http_client` source)
+- Event Data: The decoded event data. (eg: the decoded HTTP body)
+- Source Metadata: Metadata provided by the source of the event. (eg: the HTTP headers)
+- Vector Metadata: Metadata provided by Vector. (eg: the time when Vector received the event)
+
+#### Without Log Namespacing
+All three of these are placed at the root of the event. The exact layout depends on the source,
+some fields are configurable, and the [global log schema] can change the name / location of some
+fields.
+
+Example event from the `datadog agent logs` source (with JSON decoder)
+
+```json
+{
+  "ddsource": "vector",
+  "ddtags": "env:prod",
+  "hostname": "alpha",
+  "foo": "foo field",
+  "service": "cernan",
+  "source_type": "datadog_agent",
+  "bar": "bar field",
+  "status": "warning",
+  "timestamp": "1970-02-14T20:44:57.570Z"
+}
+```
+
+#### With Log Namespacing
+When enabled, the layout of this data is well-defined and consistent.
+
+Event Data (and _only_ Event Data) is placed at the root of the event (eg: `.`).
+Source metadata is placed in event metadata, prefixed by the source name. (eg: `%datadog_agent`)
+Vector metadata is placed in event metadata, prefixed by `vector`. (eg: `%vector`)
+
+Generally sinks will only send the event data. If you want to include any metadata fields,
+it's recommended to use a [remap] transform to add data to the event as needed.
+
+It's important to note that previously the type of an event (`.`) would always be an object
+with fields. Now it is possible for event to be any type, such as a string.
```

Comment:
Currently in Remap, even with Log Namespacing, if you assign `.` to a non object or array value it will place that value under the `message` field.

We should either change it so it assigns the value to root in the Vector namespace, or call that out in the docs here. I'm in favour of the former.

Reply:
I'm working on making this change. Will have a new PR up shortly.

Reply:
https://github.com/vectordotdev/vector/pull/17807